[expression] add point_on_surface function

nirvn opened the pull request:
@m-kuhn @nyalldawson as discussed, here's a point_on_surface geometry function for the expression engine. 

Here's a screenshot of centroid vs point_on_surface (made possible thanks to the geometry generator feature):
![pos](https://cloud.githubusercontent.com/assets/1728657/11734497/8dda1b60-9fea-11e5-9a8d-80fa5151ea7e.png)

nirvn committed:
[expression] add point_on_surface function

nyalldawson commented:
@nirvn thanks! as discussed, can you add tests for line, point and null/non geometry inputs too?

nirvn commented:
@nyalldawson yeah, I'll do that over the weekend.

m-kuhn commented:
Would also be nice to add some more words to the help document concerning what differntiates it from centroid and what it does with multipolygons.

nirvn commented:
@nyalldawson , I've added tests for line and point for centroid and point_on_surface in this updated commit (https://github.com/nirvn/QGIS/commit/2369e431e0ea23f340d62c2143f5352ed95893d0). I wasn't sure hw to add a non geometry input (i.e. not sure what I should have checked the return value against), so I'll have to leave this one to you, sorry :/

@m-kuhn I've updated the description to match what postgis uses. It's similar, but uses the word "guaranteed to be", which is the key difference from centroid. I didn't add anything for multipolygons as the function always return a single point, whatever geometry is being used as input.

nirvn commented:
Superseded by PR #2570 , closing.
